FRC1076-2016-Competition/src/org/usfirst/frc/team1076/robot/physical/AutoSelector.java

```diff
@@ -16,26 +16,71 @@
 	public AutoSelector() {
 		//button = new Button();
 		SmartDashboard.putNumber("Autonomous", 1);
+		SmartDashboard.putBoolean("Autonomous Backwards", false);
 	}
 
-	public AutoState getState() {
+	public AutoState getState(IRobot robot) {
 		int chooser = (int) SmartDashboard.getNumber("Autonomous", 1);
-		IChannel channel = new Channel(5880);
+		boolean backwards = SmartDashboard.getBoolean("Autonomous Backwards", false);
+		AutoState chain = new NothingAutonomous();
 		switch(chooser) {
-		// TODO: make the actual state chains
+		// TODO: Refine and test the state chains.
 		case 1:
-			return new UDPAutonomous(channel, FieldPosition.Left)
-				.setNext(new ForwardAutonomous(100, 0.2));
+			if (!backwards) {

```

Comment:
I think you have some of this backwards. We should be turning around when we're backwards, and also have negative speed.